evalml/tests/automl_tests/test_automl.py

```diff
@@ -2180,3 +2180,16 @@ def test_automl_pipeline_params_kwargs(mock_fit, mock_score, X_y_multi):
         if 'Decision Tree Classifier' in row['parameters']:
             assert 0.1 < row['parameters']['Decision Tree Classifier']['ccp_alpha'] < 0.5
             assert row['parameters']['Decision Tree Classifier']['max_depth'] == 2
+
+
+@pytest.mark.parametrize("random_state", [0, 1, 9, np.random.RandomState(100)])
+@patch('evalml.pipelines.MulticlassClassificationPipeline.score')
+@patch('evalml.pipelines.MulticlassClassificationPipeline.fit')
+def test_automl_pipeline_random_state(mock_fit, mock_score, random_state, X_y_multi):
+    X, y = X_y_multi
+    automl = AutoMLSearch(X_train=X, y_train=y, problem_type='multiclass', random_state=random_state, n_jobs=1)
+    automl.search()
+    if isinstance(random_state, int):
+        random_state = automl.random_state
```

Comment:
@bchen1116 I thought the plan was to create the pipeline with `random_state=4` if the user passed in 4 as the random state to `AutoMLSearch`. This is creating a pipeline with `np.random.RandomState(4)`, no?

Reply:
@freddyaboulton the end result here is the same, right? If the user passes in `random_state=4` to AutoMLSearch, we set the `random_state` of AutoMLSearch to be `np.random.RandomState(4)`. If we initialize a pipeline with `random_state=4`, the pipeline's `random_state` will then also be initialized as `np.random.RandomState(4)` since it just uses the int as a seed. 

I used this as a shortcut since the end values would be equivalent. Let me know what you think!

Reply:
@bchen1116 My concern is that the state of `self.random_state` changes throughout the lifetime of `AutoMLSearch`. So the pipeline is not actually being initialized with `np.random.RandomState(4)`. 

![image](https://user-images.githubusercontent.com/41651716/105422826-66469300-5c12-11eb-9193-1e838787d356.png)

I think this means we should save the original value of the `random_state` parameter before `random_seed` is created. I think this is another reason why I don't like `np.random.RandomState` hehehe

Reply:
Got it, this makes sense. Thanks for the clarification!

Reply:
Good catch @freddyaboulton.  I'll re-review after you adjust, Bryan.